src/setup.ts

```diff
@@ -51,6 +53,64 @@ const labelNodes = () => {
 
 const setup = () => window.onload = () => {
     labelNodes();
+    let comments: Promise<NTRComment[]> = getComments();
+    let lastSelected: HTMLElement | undefined = undefined;
+    let lastSelectedBackground: string = "";
+
+    comments.then((com: NTRComment[]) => {
+        com.forEach(element => {
+            let elem: HTMLElement | null = document.getElementById(element.nodeID);
+            let i: number = 0;
+
+            // doesn't work very well for displaying more than one comment and i don't really know why
+            if (elem) {
+                elem.innerHTML = elem.innerHTML.substring(0, element.startOffset) +
+                    "<span id=comment_text_" + i + " onMouseEnter=\"document.getElementById(&quot;comment_" + i +
+                    "&quot;).style.display = &quot;block&quot;\" onMouseLeave=\"document.getElementById(&quot;comment_" + i +
+                    "&quot;).style.display = &quot;none&quot;\">" +
+                    elem.innerHTML.substring(element.startOffset, element.endOffset) +
+                    "</span><div id=comment_" + i + " style=\"background-color: none; display: none; text-align: center;\">" +
+                    element.text + "</div>" +
+                    elem.innerHTML.substring(element.endOffset);
```

Comment:
This is pretty hard to read - it may be a better approach to add just an empty element with a special id, and then attach a React component for comments to it. Then it'll be much more easily readable, more easily modifiable (epecially with onMouseEnter/onMouseLeave events, but those should just be CSS), and more easily testable.